src/modules/hooks/base_hooks.js

```diff
@@ -0,0 +1,55 @@
+module.exports = class BaseHooks {
+  constructor() {
+    this._events = {};
```

Comment:
```suggestion
    this._events = new Map();
```

Please use map for this, and since we aren't using TypeScript please add a comment for what's the type of key and value, for example:
```
/*
 * Storing a map from event name to Set of callbacks
 */
```